### PR TITLE
observeNodes and unobserveNodes

### PR DESCRIPTION
**Pull request by rictic**
Source: https://github.com/Polymer/polymer/blob/9cd6b796a1f9d3232d9f36facb6d8118913a6794/src/lib/dom-api.html#L110

### CODE REVIEW

File: polymer.externs.js [hunk 1]
```diff
@@ -571,6 +571,46 @@ PolymerDomApi.prototype.insertBefore = function(node, beforeNode) {};
...
+ * })}
+ */
+PolymerDomApi.ObserveCallback;
+
+/**
+ * A virtual type for observer callback handles.
+ *
+ * @private @constructor
+ */
+PolymerDomApi.ObserveHandle = function() {};
```
Comment (jklein24) [inline polymer.externs.js]: This might make more sense as a simple typedef. WDYT?

File: polymer.externs.js [hunk 2]
```diff
@@ -571,6 +571,46 @@ PolymerDomApi.prototype.insertBefore = function(node, beforeNode) {};
...
+ * Notifies callers about changes to the element's effective child nodes,
+ * the same list as returned by `getEffectiveChildNodes`.
+ *
+ * @param {PolymerDomApi.ObserveCallback} callback The supplied callback
+ * is called with an `info` argument which is an object that provides
+ * the `target` on which the changes occurred, a list of any nodes
+ * added in the `addedNodes` array, and nodes removed in the
+ * `removedNodes` array.
+ *
+ * @return {PolymerDomApi.ObserveHandle} Handle which is the argument to
```
Comment (jklein24) [inline polymer.externs.js]: `!PolymerApi.ObserveHandle` here and elsewhere.

File: polymer.externs.js [hunk 3]
```diff
@@ -571,6 +571,46 @@ PolymerDomApi.prototype.insertBefore = function(node, beforeNode) {};
...
+ *
+ * @private @constructor
+ */
+PolymerDomApi.ObserveHandle = function() {};
+
+/**
+ * Notifies callers about changes to the element's effective child nodes,
+ * the same list as returned by `getEffectiveChildNodes`.
+ *
+ * @param {PolymerDomApi.ObserveCallback} callback The supplied callback
```
Comment (jklein24) [inline polymer.externs.js]: Should this be nullable?

File: polymer.externs.js (hunk 1 above)
Comment (rictic) [inline polymer.externs.js]: I designed it this way so that it was a nominal type that the user can't create themselves.

**Comment by rictic** [comment]
Comments addressed, PTAL!

**Comment by jklein24** [comment]
LGTM

**Pull request merged**